compiler/src/dmd/escape.d

```diff
@@ -2234,26 +2275,29 @@ public void findAllOuterAccessedVariables(FuncDeclaration fd, VarDeclarations* v
     }
 }
 
+// Maps `sequenceNumber` of a `VarDeclaration` to an object that contains the
+// reason it failed to infer `scope`
+// https://issues.dlang.org/show_bug.cgi?id=23295
+private __gshared RootObject[int] scopeInferFailure;
```

Comment:
Having this inside a global instance of a struct called inferenceState or something to that effect would be good

Reply:
Done

Reply:
cheers